src/Thinktecture.Relay.Server.Protocols.SignalR/ConnectorHub.cs

```diff
@@ -72,10 +73,18 @@ public class ConnectorHub<TRequest, TResponse, TAcknowledge> : Hub<IConnector<TR
 		/// <inheritdoc />
 		public override async Task OnConnectedAsync()
 		{
-			var tenant = Context.User.GetTenantInfo();
+			var tenant = Context.User?.GetTenantInfo();
+
+			if (tenant == null)
+			{
+				_logger.LogError("Rejecting incoming connection {ConnectionId} because of missing tenant info", Context.ConnectionId);
+				Context.Abort();
+				return;
+			}
+
 			_logger.LogDebug("Connection {ConnectionId} incoming for tenant {@Tenant}", Context.ConnectionId, tenant);
 
-			await _connectorRegistry.RegisterAsync(Context.ConnectionId, tenant.Id, Context.GetHttpContext().Connection.RemoteIpAddress);
+			await _connectorRegistry.RegisterAsync(Context.ConnectionId, tenant.Id, Context.GetHttpContext()!.Connection.RemoteIpAddress);
```

Comment:
Maybe a fallback to IP `0.0.0.0` instead of non-null assertion?

Reply:
Remote IP address is optional on the registry. We could pass null.